### PR TITLE
Fix NPE on Boat#getStatus

### PR DESCRIPTION
**Pull request by LemonCaramel**
Tested in the three situations

TestPlugin.java
```java
@Override
public void onEnable() {
    this.getServer().getPluginManager().registerEvents(this, this);

    /*
     * Get status at the same time as entity spawn
     */
    this.getServer().getCommandMap().register("boat", new Command("boat") {
        @Override
        public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
            if (sender instanceof Player player) {
                final var loc = player.getLocation();
                player.sendRawMessage( loc.getWorld().spawn(loc, Boat.class).getStatus().name() );
            }
            return false;
        }
    });
}

@EventHandler
void onEntityAddToWorld(final EntityAddToWorldEvent event) {
    if (event.getEntity() instanceof Boat boat) {
        this.logger.info("EntityAddToWorldEvent: " + boat.getStatus().name());
    }
}

/*
 * Ride on the boat and reconnect the server
 */
@EventHandler
void onEntityMount(final EntityMountEvent event) {
    if (event.getMount() instanceof Boat boat) {
        this.logger.info("EntityMountEvent: " + boat.getStatus().name());
    }
}
```

### CODE REVIEW

**Comment by Machine-Maker** [comment]
My only concern, which I mentioned in the issue I opened about this a while ago https://github.com/PaperMC/Paper/issues/8661, is that it's not immediately clear to me that it's safe to call getStatus when the boat hasn't been "in the world". It calls a lot of logic about the current position of the boat and world state which could totally not be written to work when the boat entity isn't in the world. 

An alternative solution would be to add a new "status" like "UNKNOWN" or "NOT_IN_WORLD" or something that it defaults to before it's been ticked. I am not advocating for either solution, it is just something that has to be considered (if you haven't already).

**Comment by LemonCaramel** [comment]
I want to get the always correct status returned. so I want to avoid adding a new status.
_(I want to keep this way until find a fatal problem)_


I try several tests _(change the dimension, spawn very far away, and more...)_, i have not encountered any errors or incorrect results.
```java
final Boat boat = loc.getWorld().spawn(loc, Boat.class, ett -> {
    LOGGER.info("Before: " + ett.getStatus().name());
});

LOGGER.info("After: " + boat.getStatus().name());

Bukkit.getScheduler().runTaskLater(TestPlugin.this, () -> {
    LOGGER.info("after tick(): " + boat.getStatus().name());
}, 2L);
```
 Is there anything else I should test?

**Comment by talwat** [comment]
I'm also experiencing this issue, so I'm glad there's a PR for it.

**Review by electronicboy — APPROVED** [review]

File: patches/server/0973-Fix-NPE-on-Boat-getStatus.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
++        // Paper start
++        final var handle = this.getHandle();
++        if (handle.status == null) {
++            if (handle.valid && !handle.updatingSectionStatus) {
++                handle.status = handle.getStatus();
++            } else {
++                return Status.NOT_IN_WORLD;
++            }
++        }
++        if (true) return CraftBoat.boatStatusFromNms(handle.status);
```
Comment (Owen1212055) [inline patches/server/0973-Fix-NPE-on-Boat-getStatus.patch]: Is this if (True) needed here? As isn't this just doing the same thing?

Comment (LemonCaramel) [reply patches/server/0973-Fix-NPE-on-Boat-getStatus.patch]: > Is this if (True) needed here? As isn't this just doing the same thing?

I just didn't want to call getHandle twice.

Comment (Warriorrrr) [reply patches/server/0973-Fix-NPE-on-Boat-getStatus.patch]: getHandle is just a cast so avoiding calling it twice isn't too important

**Review by kennytv — APPROVED** [review]
Removed the setting of the status field

**Pull request merged**